src/sharedInterfaces/telemetry.ts

```diff
@@ -56,3 +56,30 @@ export enum TelemetryActions {
     LoadRecentConnections = "LoadRecentConnections",
     LoadAzureSubscriptions = "LoadAzureSubscriptions",
 }
+
+export enum ActivityStatus {
+    Succeeded = "Succeeded",
+    Failed = "Failed",
+    Canceled = "Canceled",
+}
+
+export type FinishActivity = (
+    activityStatus: Exclude<ActivityStatus, ActivityStatus.Failed>,
+    additionalProperties?: Record<string, string>,
+    additionalMeasurements?: Record<string, number>,
+) => void;
+
+export type FinishActivityFailed = (
+    error?: Error,
+    includeErrorMessage?: boolean,
+    errorCode?: string,
+    errorType?: string,
+    additionalProperties?: Record<string, string>,
+    additionalMeasurements?: Record<string, number>,
+) => void;
+
+export type FinishActivityObject = {
```

Comment:
`ActivityObject` might be better so that we can add more to it than just ending the activity.